### ISSUE TITLE
Train inception v3 on a custom dataset from scratch

### ISSUE THREAD

**Issue by mpeniak** [tensorflow/models#967]
Hi guys,

I have been trying to train inception v3 models from scratch on a custom dataset. I am aware that there are some tutorials on how to fine-tune on custom data but this is not what I am looking for.

I have managed to create a new set of TFRecords for my data and then I would use the following command with the assumption it'd train the inception network.

`python train_image_classifier.py --train_dir=/tmp/train --dataset_split_name=train --dataset_dir=/notebooks/ActionRecogniser/UCF101_processed --model_name=inception_v3 --learning_rate 0.05`

However, I have just noticed that the labels.txt is copied to my dataset root folder and contains lable/category mapping for imagenet, which is not what I want. I have 101 categories and my own lables that I provided to TFRecords generation script.

I have gone through the source-code and realised that this training does not work on custom data and only on limited number of data supported in the codebase, e.g. imagenet or cifar. Therefore, my question is: how do I train the inception v3 on a custom dataset with 101 classes from scratch?

Thank you for your help!

**Comment by ogail** [comment]
If you've your own dataset then you need to develop data provider and add it to slim src code in order to use the train_image_classifier.py. Initially you need to:
- Add your data provider to [datasets folder](https://github.com/tensorflow/models/tree/master/slim/datasets)
- edit [dataset factory](https://github.com/tensorflow/models/blob/master/slim/datasets/dataset_factory.py#L30) to have mapping for your data provider.

For a working example, check [this](https://github.com/ogail/models/blob/master/slim/scripts/finetune_inception_v4_on_asins.sh)

**Comment by mpeniak** [comment]
Thanks mate, very much appreciated!

**Comment by aselle** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by wangg12** [comment]
@ogail The links you provided are dead, where can I find working examples?

**Comment by ogail** [comment]
@wangg12 updated

**Comment by foromer4** [comment]
Hi, following the above suggestions, I am running the **train_image_classifier.py** script and get this error:
`InvalidArgumentError (see above for traceback): Input to reshape is a tensor with 36100 values, but the requested shape has 784`

any ideas? thanks

**Comment by vinnitu** [comment]
gggggggjsjsjsjjjпривет

**Comment by hiroshiperera** [comment]
@ogail The links you provided are dead, can you please again update them?

**Comment by rpetar** [comment]
New links:
[datasets folder](https://github.com/tensorflow/models/tree/master/research/slim/datasetsl)
[dataset factory](https://github.com/tensorflow/models/blob/master/research/slim/datasets/dataset_factory.py#L30)

"You can start with these files and customize them for your use case, check [this](https://github.com/tensorflow/models/blob/master/research/slim/datasets/flowers.py) and [this](https://github.com/tensorflow/models/blob/master/research/slim/datasets/dataset_factory.py#L28)"
